File: scala/0509-fibonacci-number.scala
object Solution {
    var gen: Boolean = false;
    var fibonacci = List(0,1)
    def fib(n: Int): Int = {
        generate
        fibonacci(n)
    }
    
    def generate = {
        if(!gen){
            gen = true
            List.range(2, 31).map(x =>
                fibonacci = fibonacci :+ fibonacci(x-1) + fibonacci(x-2)
            )
        }
    }
}

File: swift/0509-fibonacci-number.swift
/**
 * Question Link: https://leetcode.com/problems/fibonacci-number/
 */

class Solution {
    func fib(_ n: Int) -> Int {
        let goldenRatio = (1 + 5.0.squareRoot()) / 2.0
        return Int(round(pow(goldenRatio, Double(n)) / 5.0.squareRoot()))
    }
}